### PR TITLE
Hub now offers armor set printing after the HWP is unlocked

### PR DESCRIPTION
**Pull request by carlarctg**
#### Summary
Features "Hub now offers armor set printing after HWP is unlocked"

#### Purpose of change

It's quite annoying to need to travel all the way to hub every week to see if they have what you want in stock. It shouldn't be hard to get a full set of gear, that's what they do after all. Print stuff. There was also a problem i had in my last run where the jerks never restocked on MDS after I had lost my last poncho to a zombie horde eating my npc.

#### Describe the solution

After the HWP is unlocked for printing, hub now sells sets of armor. The rig, prototype, kinetic, ballistic, turnout. Military after that's unlocked. Respectively, they cost 3, 2, 4, 4, 4, 12 for the military set, 'cause it's expensive.

#### Describe alternatives you've considered

Increasing costs. They're based on the purchase value of prototype armor, but I dunno.

Having hub sell full sets of armor rig at all times, and one armor set per shop cycle. This felt off. I like that they have surplus equipment for sale, I just don't like that I had to rely on it to get a full set.

#### Testing

Spawned myself in, spawned a hub, did the necessary missions, ordered kinetic armor for printing, fastforwarded time by one day and got my armor.

#### Additional context

Third pr, yay.

### CODE REVIEW

**Comment by andrei8l** [comment]
With this solution you have to trade for coins with the regular trade interface, then go back through the dialogue to order an armor set. You also have the price for the thing in two different places. Why not just make the armor sets guaranteed entries in the shopkeeper group?

**Comment by carlarctg** [comment]
> With this solution you have to trade for coins with the regular trade interface, then go back through the dialogue to order an armor set. You also have the price for the thing in two different places. Why not just make the armor sets guaranteed entries in the shopkeeper group?

I like the idea of hub fabricating the armor and the 24 hour wait time. I also like the concept that they're selling surplus gear in the shop group, not full sets. This is stated in alternatives. Thus, if you're a new hire, you make do with whatever they're selling, but if you've proven yourself to hub, you gain access to specialized printing of equipment, which I think is neat.

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_prototypes.json [hunk 1]
```diff
@@ -74,6 +73,161 @@
...
+                  "context": "hub_rnd"
+                }
+              ]
+            }
+          ]
+        },
+        "topic": "TALK_ROBOFAC_INTERCOM_ARMOR_FINISHED_NO"
+      },
+      {
+        "text": "How is my ordered armor going? It should be done by now.",
```
Comment (carlarctg) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_prototypes.json:188]:
```suggestion
        "text": "How is my ordered armor going?  It should be done by now.",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_prototypes.json [hunk 2]
```diff
@@ -375,6 +532,248 @@
       { "text": "I'll check in later.", "topic": "TALK_DONE" }
     ]
   },
+  {
+    "id": [ "TALK_ROBOFAC_INTERCOM_ARMOR_REQUEST" ],
+    "//": "buying armor",
+    "type": "talk_topic",
+    "dynamic_line": "Understood. The parts will finish fabrication in 24 hours, once you hand over the stipulated HGC. Depending on our currently available selection, the price will range from 2 to 4 HGC for non-specialty equipment.",
```
Comment (carlarctg) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_prototypes.json:539]:
```suggestion
    "dynamic_line": "Understood.  The parts will finish fabrication in 24 hours, once you hand over the stipulated HGC.  Depending on our currently available selection, the price will range from 2 to 4 HGC for non-specialty equipment.",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_prototypes.json [hunk 3]
```diff
@@ -375,6 +532,248 @@
...
+        "topic": "TALK_ROBOFAC_INTERCOM_SERVICES"
+      },
+      { "text": "I want to talk about something else.", "topic": "TALK_ROBOFAC_INTERCOM_SERVICES" },
+      { "text": "I'll check in later.", "topic": "TALK_DONE" }
+    ]
+  },
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_ARMOR_FINISHED_NO",
+    "type": "talk_topic",
+    "dynamic_line": "The parts are still being printed. Please be patient.",
```
Comment (carlarctg) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_prototypes.json:680]:
```suggestion
    "dynamic_line": "The parts are still being printed.  Please be patient.",
```

File: data/json/npcs/robofac/robofac_intercom/robofac_intercom_prototypes.json [hunk 4]
```diff
@@ -375,6 +532,248 @@
...
+    "responses": [ { "text": "Okay, I'll check in later.", "topic": "TALK_ROBOFAC_INTERCOM_SERVICES" } ]
+  },
+  {
+    "id": "TALK_ROBOFAC_INTERCOM_ARMOR_FINISHED_YES",
+    "type": "talk_topic",
+    "dynamic_line": "Yes, we've finished printing the requested armor pieces. Transferring to you now.",
```
Comment (carlarctg) [inline data/json/npcs/robofac/robofac_intercom/robofac_intercom_prototypes.json:686]:
```suggestion
    "dynamic_line": "Yes, we've finished printing the requested armor pieces.  Transferring to you now.",
```

**Comment by John-Candlebury** [comment]
Seems good to me, conceptually.

**Pull request merged**